### ISSUE TITLE
Re-order event functions

### ISSUE THREAD

**Issue by jondashkyle** [ungoldman/choo-store#1]
This is awesome! Only thing which stood out to me is the prop ordering differs from choo. `state, emitter` (choo) vs `emitter, state` (choo-store). What are thoughts on restructuring that, and perhaps exposing `app` as well?

```js
function event (options, store, state, emitter, app) { }
```

Really awesome work!

**Comment by ungoldman** [comment]
Hey @jondashkyle! So in the context of `choo-store`, `store` represents `state` *local to this store*. The idea being it encourages manipulating local store vs. using events to manipulate state in other places. Another way to look at it would be:

```js
function event (options, state, emitter, globalState) {}
```

I also like this ordering because it makes it easier to drop global state as a param when not needed:

```js
(opts, store, emitter) => {
  store.foo = opts.baz
  emitter.emit('render')
}
```

Having the app level state as third params means you'd have to include it every time you need the emitter regardless of whether or not you need it. Which is not terrible, just more typing.

*Another* option which deviates from choo but is more flexible is a single object parameter with named keys so you can pluck what you need and not care about argument order at all.

```js
function event ({ options, store, state, emitter, app }) {}
```

I've never used `app` in the context of a store but I do remember reading yesterday that it's made available by choo, so I agree that should be added.

Totally open to more thoughts and changes, just want to share my reasoning and see what you think.

**Comment by ungoldman** [comment]
*Another* option!

```js
function event ({ store, state, emitter, app }, ...eventArgs)
```

I think above is most flexible, since any number of arguments can come from nanobus and we don't want to make that more difficult to process, and parameter ordering for other things is made irrelevant too.

**Comment by ungoldman** [comment]
Ok released a new version (`0.5.0`) with reworked event parameters.

```
function event ({ data, store, state, emitter, app }) {}
```

I renamed `options` to `data` because it sounds a bit more type agnostic, as data could be anything. Also less letters to type.

Thanks again for your feedback!

**Comment by ungoldman** [comment]
Forgot about my second comment above (https://github.com/ungoldman/choo-store/issues/1#issuecomment-384005886) -- open to another refactor to go with that style instead, as needed.
